Bucketeer/Sources/Internal/Evaluation/EvaluationInteractor.swift

```diff
@@ -21,90 +21,106 @@ extension EvaluationInteractor {
         }
     }
-    var updateListeners: [String: EvaluationUpdateListener] = [:]
 
     func fetch(user: User, timeoutMillis: Int64?, completion: ((GetEvaluationsResult) -> Void)?) {
-        let currentEvaluationsId = self.currentEvaluationsId
-        let evaluationDao = self.evaluationDao
+
         let logger = self.logger
-        let featureTag = self.featureTag
+        let evaluatedAt = evaluationStorage.evaluatedAt
+        let userAttributesUpdated = evaluationStorage.userAttributesUpdated
+        let currentEvaluationsId = evaluationStorage.currentEvaluationsId
+        let featureTag = evaluationStorage.featureTag
         apiClient.getEvaluations(
             user: user,
             userEvaluationsId: currentEvaluationsId,
-            timeoutMillis: timeoutMillis) { [weak self] result in
-                switch result {
-                case .success(let response):
-                    let newEvaluationsId = response.userEvaluationsId
-                    if currentEvaluationsId == newEvaluationsId {
-                        logger?.debug(message: "Nothing to sync")
-                        completion?(result)
-                        return
-                    }
-                    let newEvaluations = response.evaluations.evaluations
-                    do {
-                        try evaluationDao.deleteAllAndInsert(userId: user.id, evaluations: newEvaluations)
-                    } catch let error {
-                        logger?.error(error)
-                        completion?(.failure(error: .init(error: error), featureTag: featureTag))
-                        return
-                    }
-                    self?.currentEvaluationsId = newEvaluationsId
-                    self?.evaluations[user.id] = newEvaluations
-
-                    // Update listeners should be called on the main thread
-                    // to avoid unintentional lock on Interactor's execution thread.
-                    DispatchQueue.main.async {
-                        self?.updateListeners.forEach({ _, listener in
-                            listener.onUpdate()
-                        })
-                    }
-
-                    completion?(result)
-                case .failure:
+            timeoutMillis: timeoutMillis,
+            condition: UserEvaluationCondition(
+                evaluatedAt: evaluatedAt,
+                userAttributesUpdated: userAttributesUpdated)) { [weak self] result in
+            switch result {
+            case .success(let response):
+                let newEvaluationsId = response.userEvaluationsId
+                let evaluatedAt = response.evaluations.createdAt
+                let newEvaluations = response.evaluations.evaluations
+                if currentEvaluationsId == newEvaluationsId {
+                    logger?.debug(message: "Nothing to sync")
                     completion?(result)
+                    return
+                }
+
```

Comment:
The server might send the evaluations as null if there is nothing to sync.
```suggestion
                let newEvaluationsId = response.userEvaluationsId
                if currentEvaluationsId == newEvaluationsId {
                    logger?.debug(message: "Nothing to sync")
                    completion?(result)
                    return
                }
                let evaluatedAt = response.evaluations.createdAt
                let newEvaluations = response.evaluations.evaluations
```

Reply:
Your suggestion will do the same as the original code.
But It's okay to apply. 
Updated 
https://github.com/bucketeer-io/ios-client-sdk/pull/19/commits/af1091d63dc4fabd766176f4913f26639a40b9dc